Description: sets $grub_prefix as in grub-install on NetBSD and OpenBSD.

--- util/grub-mkconfig.in.orig	2010-01-02 14:42:38.000000000 +0100
+++ util/grub-mkconfig.in
@@ -24,10 +24,10 @@ sbindir=@sbindir@
 libdir=@libdir@
 sysconfdir=@sysconfdir@
 package_version=@PACKAGE_VERSION@
+host_os=@host_os@
 datarootdir=@datarootdir@
 datadir=@datadir@
 pkgdatadir=${datadir}/`echo @PACKAGE_TARNAME@ | sed "${transform}"`
-grub_prefix=`echo /boot/grub | sed ${transform}`
 grub_cfg=""
 grub_mkconfig_dir=${sysconfdir}/grub.d
 
@@ -75,6 +75,18 @@ done
 
 . ${libdir}/grub/grub-mkconfig_lib
 
+case "$host_os" in
+netbsd* | openbsd*)
+    # Because /boot is used for the boot block in NetBSD and OpenBSD, use /grub
+    # instead of /boot/grub.
+    grub_prefix=`echo /grub | sed ${transform}`
+    ;;
+*)
+    # Use /boot/grub by default.
+    grub_prefix=`echo /boot/grub | sed ${transform}`
+    ;;
+esac
+
 if [ "x$EUID" = "x" ] ; then
   EUID=`id -u`
 fi
